components/sup/src/main.rs

```diff
@@ -284,6 +284,13 @@ fn get_ring_key(m: &ArgMatches) -> Result<Option<SymKey>> {
     }
 }
 
+fn get_gossip_listen_addr(m: &ArgMatches) -> Result<GossipListenAddr> {
+    match m.value_of("LISTEN_GOSSIP") {
+        Some(addr_str) => GossipListenAddr::from_str(addr_str),
+        None => Err(sup_error!(Error::IPFailed)),
```

Comment:
Since you've got a default value for `LISTEN_GOSSIP`, it should be impossible to get `None` here, right? If so, we could simplify this to something like 

```rust
GossipListenAddr::from_str(m.value_of("LISTEN_GOSSIP").expect("LISTEN_GOSSIP should always have a value"))
```

Reply:
Yeah, I like that better. I've been trying to stay away from using `expect` and `unwrap` outside of tests, but that does feel better than the match I'm using.

Reply:
Seeing `unwrap` in production code is often a red flag because it will cause a panic if the thing isn't actually unwrappable. It is OK in some cases, though, when you are certain (because of other code around) that it can always be unwrapped (e.g., you ascertained that `thing.is_some()` returns `true` before you run `thing.unwrap()`. Usually we'll call that out with a comment in our code.

`expect` does the same thing as `unwrap`, but allows you to add a customized message string. More importantly, though, it also gives you the location in code where `expect` was called; due to internal code shenanigans, `unwrap` _will not_ do this, instead giving you the location of `unwrap` itself in the standard library code, which is useless.

Here, you could legitimately use either one because of the guarantees that clap provides, given how we've configured things. I opted for `expect`, though, because it would provide a useful message if at some point we ever removed the default value from the clap configuration, and then started getting unexpected panics.